controllers/storagecluster/provider_server.go

```diff
@@ -304,6 +304,14 @@ func GetProviderAPIServerDeployment(instance *ocsv1.StorageCluster) *appsv1.Depl
 							},
 						},
 					},
+					Tolerations: []corev1.Toleration{
+						{
+							Effect:   corev1.TaintEffectNoSchedule,
+							Key:      "node.ocs.openshift.io/storage",
```

Comment:
yes you are right, These taints are mainly added by the customers manually on the nodes if I am not mistaken.

Reply:
@iamniting the configuration above is the one we use in the MS when deploying ODF. I think the concern is that the change you made does not take into consideration that a user will deploy with different taints. 

Can you please comment on that?

Reply:
yes you are correct @nb-ohad

Reply:
@iamniting for me this changes breaks if someone doesn't set taints on the nodes or run odf on all worker nodes example ODF managed service in provider mode.

Reply:
After discussing with @iamniting this is not a breaking change but again it doesn't solve the BZ as the pod has the toleration doesn't ensure that it will not come up on the infra node (which is the problem mentioned in BZ).